staging/src/k8s.io/cli-runtime/pkg/resource/visitor.go

```diff
@@ -425,18 +427,25 @@ func (v FlattenListVisitor) Visit(fn VisitorFunc) error {
 		if info.Mapping != nil && !info.Mapping.GroupVersionKind.Empty() {
 			preferredGVKs = append(preferredGVKs, info.Mapping.GroupVersionKind)
 		}
-
+		errs := []error{}
 		for i := range items {
 			item, err := v.mapper.infoForObject(items[i], v.typer, preferredGVKs)
 			if err != nil {
-				return err
+				errs = append(errs, err)
 			}
 			if len(info.ResourceVersion) != 0 {
 				item.ResourceVersion = info.ResourceVersion
 			}
 			if err := fn(item, nil); err != nil {
-				return err
+				errs = append(errs, err)
+			}
+		}
+
+		if len(errs) != 0 {
```

Comment:
You don't need these checks, you can just `return utilerrors.NewAggregate(errs)` since it returns `nil` if the list is empty (see https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go#L36-L40)